### ISSUE TITLE
CLI Project Creation - Issue with the AppHost.Program.cs 

### ISSUE THREAD

**Issue by AClerbois** [dotnet/aspire#1078]
Hi, 

I play with your new revolution named Aspire. I found a small bug. 

When we use a project name containing a "." (dot) the file Program.cs from AppHost's project don't point to the correct project reference:

```csharp 

var apiservice = builder.AddProject<Projects.Sot.SessionSphere_ApiService>("apiservice");

builder.AddProject<Projects.Sot.SessionSphere_Web>("webfrontend")
    **.WithReference(apiservice);
```

And should : 
```csharp 
var apiservice = builder.AddProject<Projects.Sot_SessionSphere_ApiService>("apiservice");

builder.AddProject<Projects.Sot_SessionSphere_Web>("webfrontend")
    **.WithReference(apiservice);
```

## How to reproduce 

1. Let's create a new starter via the dotnet new template. 

```ps
 dotnet new aspire-starter -n Sot.SessionSphere
```

2. Open the file and observe the template has failed to use the first separator.

**Comment by DamianEdwards** [comment]
This was fixed along with the fix for #945 and will be in preview.2 thanks!

**Comment by AClerbois** [comment]
@DamianEdwards awesome news. Thanks!
